src/components/logger/logger.js

```diff
@@ -53,26 +55,26 @@ export default class Logger extends React.Component {
   };
 
   state = {
+    logs: [],
     events: [],
   };
 
   onClear = () => this.setState({ events: [] });
+  onEvent = (event) => this.setState((state) => ({ events: [...state.events, event] }));
 
-  componentWillUpdate({ logs }) {
-    this.logsSubscriptions.map((sub) => sub.unsubscribe());
-    this.logsSubscriptions = logs.map((obs) => {
-      return obs.subscribe((event) => {
-        this.setState((state) => ({ events: [...state.events, event] }));
-      });
-    });
+  componentWillUpdate(nextProps) {
+    const { logs: prevLogs } = this.state;
+    const { logs: nextLogs } = nextProps;
+
+    if (!isEqual(prevLogs, nextLogs)) {
+      this.sub.unsubscribe();
+      this.sub = merge(...nextLogs).subscribe(this.onEvent);
```

Comment:
What is exactly inside `nextLogs` ? I think that it should not be a array, it's only one subscription.
It won't be like `[logs: ServiceA.method$(), ServiceB.method$()]` but more like `invoker.logs()`.
There will be only one subscription to invoker service logs method that will contains the requests/responses events for all services.

Reply:
@stephanebenayoun logs: Observable<Event>[]

Ok, so I will replace it by one sub.